cli/azd/pkg/project/framework_service_docker.go

```diff
@@ -200,6 +202,22 @@ func (p *dockerProject) Build(
 		return &ServiceBuildResult{Restore: restoreOutput}, nil
 	}
 
+	// if it's a java project without Dockerfile, we help to package jar and add a default Dockerfile for Docker build
+	if serviceConfig.Language == ServiceLanguageJava && serviceConfig.Docker.Path == "" {
+		mvnCli := maven.NewCli(exec.NewCommandRunner(nil))
+		err := mvnCli.CleanPackage(ctx, serviceConfig.RelativePath, serviceConfig.Project.Path)
+		if err != nil {
+			return nil, err
+		}
+		defaultDockerfilePath, err := addDefaultDockerfileForJavaProject(serviceConfig.Name)
+		if err != nil {
+			return nil, err
+		}
+		serviceConfig.Docker = DockerProjectOptions{
+			Path: defaultDockerfilePath,
+		}
+	}
```

Comment:
I think you're running into https://github.com/microsoft/Oryx/issues/2562 here.

In the meantime, I think we can consider testing out Paketo's Buildpack implementation. We can change the `build` on line 466 to point to `paketobuildpacks/builder-jammy-base` and see how well this works.

Reply:
Will test it.